sdk/nodejs/output.ts

```diff
@@ -143,7 +159,7 @@ class OutputImpl<T> implements OutputInstance<T> {
         }
 
         this.resources = () => new Set<Resource>(resourcesArray);
-        this.promise = () => promise;
+        this.promise = (withUnknowns?: boolean) => value.then(getValue => getValue(withUnknowns));
```

Comment:
from the coment above, i thnk this would actually be easier to read and understand extracted out.  i.e.:

```
this.promise = async (withUnknowns?: boolean) => MyHelper(exactly, what, i, need, to, capture);
```

Doing this will have the benefit of not needing to create locals in the outer scope (like `const value`) that then have to be analyzed to see how they're actually used.